compiler-rt/test/hwasan/TestCases/memcmp.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %clangxx_hwasan -O0 %s -o %t && not %run %t 2>&1 | FileCheck %s
+// RUN: %clangxx_hwasan -O1 %s -o %t && not %run %t 2>&1 | FileCheck %s
+// RUN: %clangxx_hwasan -O2 %s -o %t && not %run %t 2>&1 | FileCheck %s
+// RUN: %clangxx_hwasan -O3 %s -o %t && not %run %t 2>&1 | FileCheck %s
+
+// REQUIRES: pointer-tagging
+
+#include <string.h>
+int main(int argc, char **argv) {
+  char a1[] = {static_cast<char>(argc), 2, 3, 4};
+  char a2[] = {1, static_cast<char>(2*argc), 3, 4};
+  int res = memcmp(a1, a2, 4 + argc);  // BOOM
+  // CHECK: HWAddressSanitizer: tag-mismatch on address
```

Comment:
I could be nice to check more output

Reply:
Can you try to remove // REQUIRES: pointer-tagging
and use malloc, instead of stack
and test on x86 host ?

Reply:
this test is likely wrong on 'memcpy(p, a, sizeof(a));' not memcmp
it should free after memcpy

that why I asking to '// CHECK:' for other events
eg. stack trace lines